new-lamassu-admin/src/pages/Blacklist/BlacklistModal.js

```diff
@@ -0,0 +1,99 @@
+import { makeStyles } from '@material-ui/core/styles'
+import * as R from 'ramda'
+import React, { useState } from 'react'
+
+import Modal from 'src/components/Modal'
+import { Link } from 'src/components/buttons'
+import { TextInput } from 'src/components/inputs'
+import { H3 } from 'src/components/typography'
+import coinValidator from 'src/utils/coins/coinValidator'
+
+import styles from './Blacklist.styles'
+const useStyles = makeStyles(styles)
+
+const BlackListModal = ({
+  showModal,
+  toggleModal,
+  selectedCoin,
+  addToBlacklist
+}) => {
+  const classes = useStyles()
+
+  const [addressField, setAddressField] = useState('')
+  const [invalidAddress, setInvalidAddress] = useState(false)
+
+  const handleChange = event => {
+    if (event.target.value === '') {
+      setInvalidAddress(false)
+    }
+    setAddressField(event.target.value)
+  }
+
+  const handleAddToBlacklist = () => {
+    if (addressField.trim() === '') {
+      setInvalidAddress(true)
+      return
+    }
+    if (coinValidator[selectedCoin.code](addressField.trim())) {
+      addToBlacklist(selectedCoin.code, addressField.trim())
+      handleClose()
+    } else {
+      setInvalidAddress(true)
+    }
+  }
```

Comment:
We don't really wanna close the modal before getting the response from the server. See how other wizards handle this.